sys/kern/kern_sig.c

```diff
@@ -320,6 +321,7 @@ siginfo_native_to_siginfo(const struct siginfo_native *si_n,
 	si->si_pid = si_n->si_pid;
 	si->si_uid = si_n->si_uid;
 	si->si_status = si_n->si_status;
+	/* XXXAR: is using $ddc correct here? */
 	si->si_addr = (__cheri_tocap void * __capability)si_n->si_addr;
 	si->si_value.sival_ptr =
 	    (__cheri_tocap void * __capability)si_n->si_value.sival_ptr;
```

Comment:
`si_addr` is supposed to be the address of the faulting instruction, so should probably be a userland capability in the `siginfo_native` form, in a sense, but an untagged capability is probably fine, since I don't imagine anyone attempts to jump to it rather than do a `sigreturn` or `siglongjmp` somewhere else.